src/test/java/seedu/address/logic/parser/ViewCommandParserTest.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_RECIPE;
+import static seedu.address.testutil.TypicalRecipes.AGLIO_OLIO;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.ViewCommand;
+
+/**
+ * Contains integration tests and unit tests for {@code ViewCommandParser}.
+ */
+public class ViewCommandParserTest {
+
+    private final ViewCommandParser parser = new ViewCommandParser();
+
+    /**
+     * Tests parse method with valid inputs.
+     * Expects appropriate ViewCommand Object to be returned.
+     */
+    @Test
+    public void parse_validArgs_returnsViewCommand() {
+        assertParseSuccess(parser, " -x 1", new ViewCommand(INDEX_FIRST_RECIPE));
+        assertParseSuccess(parser, " Aglio Olio\n", new ViewCommand(AGLIO_OLIO.getName()));
+    }
```

Comment:
can consider adding an assert to show that view command is case-insensitive

```suggestion
    @Test
    public void parse_validArgs_returnsViewCommand() {
        assertParseSuccess(parser, " -x 1", new ViewCommand(INDEX_FIRST_RECIPE));
        assertParseSuccess(parser, " Aglio Olio\n", new ViewCommand(AGLIO_OLIO.getName()));
        assertParseSuccess(parser, "aglio olio", new ViewCommand(AGLIO_OLIO.getName()));
    }
```